.github/workflows/windows.yml

```diff
@@ -0,0 +1,58 @@
+name: windows
+
+on:
+  push:
+    paths-ignore:
+    - '.github/workflows/**'
+    - '!.github/workflows/windows.yml'
+    - '!.github/workflows/EnterDevShell.ps1'
+  pull_request:
+    paths-ignore:
+    - '.github/workflows/**'
+    - '!.github/workflows/windows.yml'
+    - '!.github/workflows/EnterDevShell.ps1'
+
+jobs:
+  windows-msvc:
+    runs-on: windows-2022
+    steps:
+    - name: 'Checkout'
+      uses: actions/checkout@v2
+    - name: 'Setup Python'
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.x'
+    - name: Install Meson and Ninja
+      run: pip install meson ninja
+    - name: Enter DevShell
+      run: '.github\workflows\EnterDevShell.ps1 ${{ inputs.architecture }}'
+      shell: pwsh
+    - name: Configure with meson for WIN32 backend only
+      run: meson build
+    - name: Build and Install with ninja
+      run: ninja -C build install
+
+  windows-mingw:
+    runs-on: windows-2022
+    defaults:
+      run:
+        shell: msys2 {0}
+    steps:
+    - name: 'Checkout'
+      uses: actions/checkout@v2
+    - name: 'Setup MSYS2'
+      uses: msys2/setup-msys2@v2
+      with:
+        msystem: mingw64
+        update: false
+        pacboy: >-
+          toolchain:p
+          meson:p
+          ninja:p
+    - name: Enter DevShell
+      run: '.github\workflows\EnterDevShell.ps1 ${{ inputs.architecture }}'
+      shell: pwsh
+    - name: Configure with meson for WIN32 backend only
+      run: CC=gcc CXX=g++ meson build
+    - name: Build and Install with ninja
+      run: CC=gcc CXX=g++ ninja -C build install

```

Comment:
CXX is unused and can be dropped. There's no C++ code in tree (ignoring the android one) even if the configure.ac lists `AC_PROG_CXX` - it should be removed ... Separate MR if you feel like fixing it - thanks o/